### ISSUE TITLE
feature request: generate a build script

### ISSUE THREAD

**Issue by ghost** [mesonbuild/meson#12679]
It'd be nice to have meson generate a build shell (POSIX shell for improved portability) script for users. That way, non-developers of a project using meson don't need to have meson installed.

example: https://github.com/gittup/tup/blob/master/tup.1#L200

**Comment by eli-schwartz** [comment]
Are there examples of things other than tup (maybe even things that have decent real world use), that provide this functionality?


I have tried reading the tup manual numerous times, but cannot figure out how to use it as anything other than a "better Makefile" and that is the least part of what a build system does.

Also, the tup manual claims:
> The expected usage is in continuous integration environments that aren't compatible with tup's dependency checking (eg: if FUSE is not supported)

Which doesn't sound remotely promising. If you have to generate a unique build script for each Linux distro, and then another unique build script for each of the BSDs, that would be a disaster for usability. And needing to generate a unique build script for each commercial Unix would be legally impossible for most users to get access to

**Comment by bonzini** [comment]
I don't think this is feasible given the complexity of the logic to detect dependencies etc.

Replacing build.ninja generation with a shell script might be interesting for bootstrapping reasons, but building without meson installed is a requirement that is gladly left to Autoconf.
